src/goose/brokers/redis/batch.clj

```diff
@@ -0,0 +1,30 @@
+(ns goose.brokers.redis.batch
+  (:require
+    [goose.brokers.redis.commands :as redis-cmd]
+    [goose.defaults :as d]
+    [goose.job :as job]
+    [goose.retry :as retry]))
+
+(defn update-state [next]
+  (fn [{:keys [redis-conn] :as opts}
+       {:keys [id, batch-id] :as job}]
+    (if batch-id
+      (let [response (next opts job)
+            src (d/batch-set batch-id d/enqueued-job-set)
+            dst (d/batch-set batch-id d/successful-job-set)
+            retrying-batch-set (d/batch-set batch-id d/retrying-job-set)]
+        (try
+          (if (job/retried? job)
+            (redis-cmd/move-btn-set redis-conn retrying-batch-set dst id)
+            (redis-cmd/move-btn-set redis-conn src dst id))
+          (catch Exception ex
+            (let [failed-job (retry/set-failed-config job ex)
```

Comment:
See if this logic makes sense to you? It will reduce move-btn-set command to 1 time & move 2 if conditions above
```clojure
(let [src (if (state==error) retrying enqueued)
      dst (if (retry-count < max) retrying dead)]
  (move-btn-set redis-conn src dst)
```

Reply:
I wrote this logic thinking that for a job that failed, src can be either of enqueued/retrying & dst can be either of retrying/dead.

Reply:
Thanks for the suggestion @olttwa, it makes sense. I have incorporated the same in latest commit.